ngi_reports/utils/statusdb.py

```diff
@@ -65,38 +79,111 @@ def get_project_flowcell(self, project_id, open_date="2015-01-01", date_format="
+                ):
+                    project_flowcells[fc_name] = {
+                        "name": fc_name,
+                        "run_name": fc,
+                        "date": fc_date,
+                        "db": self.db.name,
+                    }
         return project_flowcells
 
+
 class ProjectSummaryConnection(statusdb_connection):
     def __init__(self, dbname="projects"):
         super(ProjectSummaryConnection, self).__init__()
         self.db = self.connection[dbname]
-        self.name_view = {k.key:k.id for k in self.db.view("project/project_name", reduce=False)}
-        self.id_view = {k.key:k.id for k in self.db.view("project/project_id", reduce=False)}
+        self.name_view = {
+            k.key: k.id for k in self.db.view("project/project_name", reduce=False)
+        }
+        self.id_view = {
+            k.key: k.id for k in self.db.view("project/project_id", reduce=False)
+        }
+
 
 class SampleRunMetricsConnection(statusdb_connection):
     def __init__(self, dbname="samples"):
         super(SampleRunMetricsConnection, self).__init__()
         self.db = self.connection[dbname]
 
+
 class FlowcellRunMetricsConnection(statusdb_connection):
     def __init__(self, dbname="flowcells"):
         super(FlowcellRunMetricsConnection, self).__init__()
         self.db = self.connection[dbname]
-        self.name_view = {k.key:k.id for k in self.db.view("names/name", reduce=False)}
-        self.proj_list = {k.key:k.value for k in self.db.view("names/project_ids_list", reduce=False) if k.key}
+        self.name_view = {k.key: k.id for k in self.db.view("names/name", reduce=False)}
+        self.proj_list = {
+            k.key: k.value
+            for k in self.db.view("names/project_ids_list", reduce=False)
+            if k.key
+        }
+
 
 class X_FlowcellRunMetricsConnection(statusdb_connection):
     def __init__(self, dbname="x_flowcells"):
         super(X_FlowcellRunMetricsConnection, self).__init__()
         self.db = self.connection[dbname]
-        self.name_view = {k.key:k.id for k in self.db.view("names/name", reduce=False)}
-        self.proj_list = {k.key:k.value for k in self.db.view("names/project_ids_list", reduce=False) if k.key}
+        self.name_view = {k.key: k.id for k in self.db.view("names/name", reduce=False)}
+        self.proj_list = {
+            k.key: k.value
+            for k in self.db.view("names/project_ids_list", reduce=False)
+            if k.key
+        }
+
+
+class NanoporeRunConnection(statusdb_connection):
+    def __init__(self, dbname="nanopore_runs"):
+        super(NanoporeRunConnection, self).__init__()
+        self.db = self.connection[dbname]
+        self.name_view = {k.key: k.id for k in self.db.view("names/name", reduce=False)}
+        self.proj_list = {
+            k.key: k.value
+            for k in self.db.view("names/project_ids_list", reduce=False)
+            if k.key
+        }
```

Comment:
We should probably transition to the cloudant api when writing new code.

Reply:
This is not new code, it's 2 years old 😆 But sure. I don't know where to begin with that though, I've never used it. Any tips?